EIPS/eip-9999.md

```diff
@@ -0,0 +1,447 @@
+    chunk_index: ChunkIndex
+    error: Optional[str]  # Error details if INVALID
+
+class PayloadFinalizeStatusV1(Container):
+    status: str  # "VALID", "INVALID", "SYNCING"
+    error: Optional[str]
+```
+
+## Rationale
+
+### Why incremental CALs
+
+The incremental CALs approach where chunk N applies CALs 0..N-1 to parent state provides:
+
+1. **Parallel Execution**: Multiple chunks can execute simultaneously once their CALs are available
+    - This could be achieved without incremental approach, but CALs would need to contain relevant state diffs from previous chunks. This would lead to very big chunks in the worst case scenario.
+2. **Early Validation**: Invalid blocks can be rejected at the first invalid chunk
+3. **Bounded Resources**: Each chunk independently bounded to `CHUNK_GAS_LIMIT`
+
+### Why CL-Driven Architecture?
+
+The consensus layer drives the execution layer for several key reasons:
+
+1. **Current design**: Most CL-EL interaction is currently CL driven
+2. **CAL Dependency Management**: The CL has the global view of which CALs have been received from the network and can orchestrate chunk execution in the optimal order based on availability
+3. **Natural Ordering**: The CL naturally enforces that chunks are executed only when their dependencies are met
+4. **Timing**: The CL knows whether chunks arrived timely
+
+### Why Separate CALs from Chunks?
+
+Separating Chunk Access Lists (containing state diffs) from chunks provides several benefits:
+
+1. **Independent Propagation**: CALs are typically much smaller than chunks (containing only state diffs) and propagate faster
+2. **Parallel Execution**: Multiple chunks can execute independently by applying prior CALs to reconstruct required pre-states
+3. **Modularity**: State diffs are logically separate from transaction data
+4. **Future Compatibility**: CALs can be extended with additional metadata without modifying chunk structure
+5. **Streaming Validation**: Validators can begin executing chunks as soon as required CALs arrive
+
+### Why Semantic Chunking Over Simple Fragmentation?
+
+Unlike simple byte-level fragmentation, semantic chunking:
+
+1. **Enables Streaming Validation**: Chunks are self-contained execution units that can be validated as they arrive
+2. **Bounds Resources**: Each chunk has maximum gas limits, bounding memory and CPU usage
+
+### Choice of Parameters
+
+- **`CHUNK_GAS_LIMIT = 2**24`** (16.7M gas): Balances parallelization benefits with chunk overhead. Large enough for complex transactions, small enough for bounded proving
+- **`MAX_CHUNKS_PER_BLOCK = 256`**: Supports up to 268 Mgas blocks
+- **`MAX_TRANSACTIONS_PER_CHUNK = 2**10`** (1,024): Smallest power of two that is higher than the number of transactions in a chunk full of basic ETH transfers (21,000 gas per tx)
+
+## Backwards Compatibility
+
+This EIP introduces breaking changes to:
+
+1. **Block Structure**: Replaces monolithic execution payload with chunked structure
```

Comment:
I'm not sure, it does change the execution payload but not the "block structure" (I'd call block header the block structure?)